distributionviewer/core/static/js/app/components/containers/chart-detail-container.js

```diff
@@ -19,6 +20,10 @@ class ChartDetailContainer extends React.Component {
   }
 
   componentDidMount() {
+    // Fetch metadata if needed for hoverStrings.
+    if (!store.getState().metricMetadataState.metadata.length) {
+      metricApi.getMetricMetadata();
+    }
```

Comment:
I'm not sure I understand this, but it looks interesting. Is this an alternative to adding `metadata` to `mapStateToProps` and checking for the existence of `this.props.metadata` here? Like this:

```javascript
componentDidMount() {
  if (!this.props.metadata) {
    metricApi.getMetricMetadata();
  }
}

// ...

const mapStateToProps = function(store, ownProps) {
  return {
    metric: store.metricState.metrics[parseInt(ownProps.params.metricId, 10)],
    metadata: store.metricMetadataState.metadata,
  };
};